app/code/core/Mage/Core/Model/Resource/Db/Abstract.php

```diff
@@ -400,6 +400,10 @@ public function load(Mage_Core_Model_Abstract $object, $value, $field = null)
      */
     protected function _getLoadSelect($field, $value, $object)
     {
+        $fields = $this->_getReadAdapter()->describeTable($this->getMainTable());
+        if(isset($fields[$field]) && isset($fields[$field]['DATA_TYPE']))   {
```

Comment:
it's enough to just call `isset($fields[$field]['DATA_TYPE'])`. isset is NOT throwing notice/warning when accessing non existing keys.

Reply:
you are totally correct, i hadn't noticed this before